### DISCUSSION TITLE
dirtyFields does not trigger re-render of component

### DISCUSSION THREAD

**Discussion by foxylion** [react-hook-form/react-hook-form#6338] [Q&A]
Hi,

I tried to run an effect based on the `dirtyFields` value of `formState` but that did not work, because the array itself seems to never change. Only the elements inside the arrays are modified. Is this expected?

Sample: https://codesandbox.io/s/crazy-galileo-j07te?file=/src/App.tsx

When entering something into the first name field the `isDirty` effect is called, but the `dirtyFields` only once on component creation.

Best,
Jakob

**Comment by bluebill1049** [comment]
related to this: https://github.com/react-hook-form/react-hook-form/issues/6243 because we are batching formState update.
